src/ui/window_list.c

```diff
@@ -1067,19 +1065,13 @@ wins_create_summary(gboolean unread)
     while (curr) {
         ProfWin* window = g_hash_table_lookup(windows, curr->data);
         if (!unread || (unread && win_unread(window) > 0)) {
-            GString* line = g_string_new("");
-
             int ui_index = GPOINTER_TO_INT(curr->data);
             auto_gchar gchar* winstring = win_to_string(window);
             if (!winstring) {
-                g_string_free(line, TRUE);
                 continue;
             }
-
-            g_string_append_printf(line, "%d: %s", ui_index, winstring);
-
-            result = g_slist_append(result, strdup(line->str));
-            g_string_free(line, TRUE);
+            gchar* line = g_strdup_printf("%d: %s", ui_index, winstring);
+            result = g_slist_append(result, line);
```

Comment:
I love how much it improves the code. The only minor issue I see is that cleanup place is using `free` instead of `g_free`: 
https://github.com/profanity-im/profanity/blob/5385cc28beca6513ef2ad66a405b296f1507b73c/src/ui/console.c#L467
https://github.com/profanity-im/profanity/blob/5385cc28beca6513ef2ad66a405b296f1507b73c/src/ui/console.c#L487